src/django_peertube_runner_connector/views/runner.py

```diff
@@ -0,0 +1,62 @@
+"""API Endpoints for Runners with Django RestFramework viewsets."""
+from datetime import timezone as tz
+import uuid
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils import timezone
+
+from rest_framework import status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from django_peertube_runner_connector.forms import RunnerForm
+from django_peertube_runner_connector.models import Runner, RunnerRegistrationToken
+from django_peertube_runner_connector.serializers import RunnerSerializer
+
+
+class RunnerViewSet(viewsets.GenericViewSet):
+    """Viewset for the API of the runner object."""
+
+    queryset = Runner.objects.all()
+    serializer_class = RunnerSerializer
+
+    @action(detail=False, methods=["post"], url_path="register")
+    def register(self, request):
+        """Register a PeerTube runner."""
+        try:
+            form = RunnerForm(request.data)
+            runner = form.save(commit=False)
+        except ValueError:
+            return Response({"errors": [dict(form.errors)]}, status=400)
+
+        try:
+            registration_token = RunnerRegistrationToken.objects.get(
+                registrationToken=request.data.get("registrationToken", "")
+            )
+        except ObjectDoesNotExist:
+            return Response(
+                status=status.HTTP_404_NOT_FOUND,
+                data={"message": "Registration token is invalid"},
+            )
+
+        runner.runnerToken = "ptrt-" + str(uuid.uuid4())
+        runner.ip = request.META.get("REMOTE_ADDR")
```

Comment:
This will not work behind a reverse proxy: https://stackoverflow.com/questions/4581789/how-do-i-get-user-ip-address-in-django

I don't understand why this is not in the django request object...